src/de/mod.rs

```diff
@@ -2880,7 +2880,7 @@ impl StartTrimmer {
     #[inline(always)]
     fn trim<'a>(&mut self, event: Event<'a>) -> Option<PayloadEvent<'a>> {
         let (event, trim_next_event) = match event {
-            Event::DocType(e) => (PayloadEvent::DocType(e), false),
+            Event::DocType(e) => (PayloadEvent::DocType(e), true),
```

Comment:
Could you add a comment describing, why we need trim here because this is not obvious (otherwise it would be done initially)

Reply:
The Doctype should trim events here just like the Start/End events. Only the CData/Text events should not trim (otherwise any CData/Text following them will be ignored